src/main/java/com/websafe/ccmsbe/service/PrivacyRegulationService.java

```diff
@@ -0,0 +1,148 @@
+package com.websafe.ccmsbe.service;
+
+import com.websafe.ccmsbe.entity.PrivacyRegulation;
+import com.websafe.ccmsbe.entity.Website;
+import com.websafe.ccmsbe.repository.PrivacyRegulationRepository;
+import com.websafe.ccmsbe.repository.WebsiteRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import java.util.List;
+import java.util.Optional;
+@Service
+public class PrivacyRegulationService {
+    private final PrivacyRegulationRepository privacyRegulationRepository;
+    private final WebsiteRepository websiteRepository;
+
+    //User define exception for 'PrivacyRegulationNotFound' error
+    public static class PrivacyRegulationNotFoundException extends RuntimeException {
+        public PrivacyRegulationNotFoundException(String message) {
+            super(message);
+        }
+    }
+    //User define exception for 'WebsiteNotFound' error
+    public static class WebsiteNotFoundException extends RuntimeException {
+        public WebsiteNotFoundException(String message) {
+            super(message);
+        }
+    }
+    //User define exception for PrivacyRegulationUpdate errors
+    public static class PrivacyRegulationUpdateException extends RuntimeException {
+        public PrivacyRegulationUpdateException(String message) {
+            super(message);
+        }
+    }
+    //User define exception for PrivacyRegulationCreation errors
+    public static class PrivacyRegulationCreationException extends RuntimeException {
+        public PrivacyRegulationCreationException(String message) {
+            super(message);
+        }
+    }
+    //User define exception for PrivacyRegulationRetrieval errors
```

Comment:
```suggestion

    // User define exception for PrivacyRegulationRetrieval errors.
```

Reply:
Seems like the fix for these hasn't been pushed. Shall we push it and add a comment here?

Reply:
Fixed c4bd8fe